go/vt/vtgate/schema/update_controller_flaky_test.go

```diff
@@ -177,3 +177,153 @@ func TestMultipleUpdatesFromDifferentShards(t *testing.T) {
+		viewUpdates []string
+	}
+	type testCase struct {
+		updateViews                  []string
+		signalExpected, initExpected int
+		inputs                       []input
+		delay                        time.Duration
+		init, initFail, updateFail   bool
+	}
+	tests := []testCase{{
+		inputs: []input{{
+			shard:       "-80",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "80-",
+			viewUpdates: []string{"a"},
+		}},
+		updateViews:    []string{"a"},
+		signalExpected: 1,
+	}, {
+		inputs: []input{{
+			shard:       "0",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "0",
+			viewUpdates: []string{"b"},
+		}},
+		updateViews:    []string{"a", "b"},
+		signalExpected: 1,
+	}, {
+		inputs: []input{{
+			shard:       "0",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "0",
+			viewUpdates: []string{"b"},
+		}},
+		updateViews:    []string{"b"},
+		signalExpected: 2,
+		delay:          10 * time.Millisecond,
+	}, {
+		inputs: []input{{
+			shard: "0",
+		}, {
+			shard: "0",
+		}},
+	}, {
+		inputs: []input{{
+			shard:       "-80",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "80-",
+			viewUpdates: []string{"a"},
+		}},
+		signalExpected: 1,
+		initExpected:   1,
+		init:           true,
+	}, {
+		inputs: []input{{
+			shard:       "-80",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "80-",
+			viewUpdates: []string{"a"},
+		}},
+		signalExpected: 0,
+		initExpected:   1,
+		init:           true,
+		initFail:       true,
+	}, {
+		inputs: []input{{
+			shard:       "-80",
+			viewUpdates: []string{"a"},
+		}, {
+			shard:       "80-",
+			viewUpdates: []string{"b"},
+		}},
+		updateViews:    []string{"a", "b"},
+		signalExpected: 0,
+		updateFail:     true,
+	},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%d", i+1), func(t *testing.T) {
+			var signalNb, initNb int
+			var updatedViews []string
+			update := func(th *discovery.TabletHealth) bool {
+				updatedViews = th.Stats.ViewSchemaChanged
+				return !test.updateFail
+			}
+			signal := func() {
+				signalNb++
+			}
+			updateCont := updateController{
+				update:       update,
+				signal:       signal,
+				consumeDelay: 5 * time.Millisecond,
+				reloadKeyspace: func(th *discovery.TabletHealth) error {
+					initNb++
+					var err error
+					if test.initFail {
+						err = fmt.Errorf("error")
+					}
+					return err
+				},
+				loaded: !test.init,
+			}
+
+			for _, in := range test.inputs {
+				target := &querypb.Target{
+					Keyspace:   "ks",
+					Shard:      in.shard,
+					TabletType: topodatapb.TabletType_PRIMARY,
+				}
+				tablet := &topodatapb.Tablet{
+					Keyspace: target.Keyspace,
+					Shard:    target.Shard,
+					Type:     target.TabletType,
+				}
+				d := &discovery.TabletHealth{
+					Tablet:  tablet,
+					Target:  target,
+					Serving: true,
+					Stats:   &querypb.RealtimeStats{ViewSchemaChanged: in.viewUpdates},
+				}
+				if test.delay > 0 {
+					time.Sleep(test.delay)
+				}
```

Comment:
This might be a source of flakiness

Reply:
The test is to see that multiple signals happen because they are received at different time intervals.